src/screens/editMonitoring/hooks/useEditMonitoring.ts

```diff
@@ -0,0 +1,262 @@
+import { useSnackBar } from '../../../utils/renderSnackBar';
+import { useState, useMemo, useEffect } from 'react';
+import { TWeekDayAvailabilityState } from './types';
+import {
+  WEEK_DAY_AVAILABILITY_INITIAL_STATE,
+  END_HOURS_OPTIONS,
+  START_HOURS_OPTIONS,
+} from './contants';
+import { TAvailability } from '../../../service/requests/useUpdateMonitorAvailabilityRequest/types';
+import useGetMonitorRequest from '../../../service/requests/useGetMonitorRequest';
+import { SelectChangeEvent } from '@mui/material';
+import usePatchMonitorRequest from '../../../service/requests/usePatchMonitorRequest';
+import useGetLoggedUser from '../../../service/storage/getLoggedUser';
+import useQuery from '../../../utils/useQuery';
+import { useNavigate } from 'react-router-dom';
+import { SCREENS } from '../../../utils/screens';
+
+const useEditMonitoring = () => {
+  const navigate = useNavigate();
+  const query = useQuery();
+  const user = useGetLoggedUser();
+  const { showSuccessSnackBar, showErrorSnackBar } = useSnackBar();
+
+  const {
+    getMonitor,
+    data: monitor,
+    isLoading: isLoadingGetMonitor,
+  } = useGetMonitorRequest();
+  const {
+    isLoading: isLoadingPatch,
+    isSuccess: isSuccessPatch,
+    error: errorPatch,
+    patchMonitor,
+  } = usePatchMonitorRequest();
+
+  const [preference, setPreference] = useState('');
+  const [editMode, setEditMode] = useState(
+    query.get('edit') === 'true' ? true : false
+  );
+  const [isOpenCancelModal, setIsOpenCancelModal] = useState(false);
+  const [weekDayAvailability, setWeekDayAvailability] = useState<
+    TWeekDayAvailabilityState[]
+  >(WEEK_DAY_AVAILABILITY_INITIAL_STATE);
+
+  const resetWeekDay = () => {
+    setWeekDayAvailability(
+      weekDayAvailability.map((weekDay) => ({
+        ...weekDay,
+        isSelected: false,
+        fromHourIndex: -1,
+        toHourIndex: -1,
+      }))
+    );
+  };
```

Comment:
Sempre que o novo valor de um estado depender do anterior, você precisa fazer o set assim:

```suggestion
    setWeekDayAvailability((previousValue) => (
      previousValue.map((weekDay) => ({
        ...weekDay,
        isSelected: false,
        fromHourIndex: -1,
        toHourIndex: -1,
      }))
    ));
  };
```